ruby/README.md

```diff
@@ -0,0 +1,130 @@
+# OpenTelemetry Lambda Ruby
+
+Scripts and files used to build AWS Lambda Layers for running OpenTelemetry on AWS Lambda for Ruby.
+
+Requirement:
+* [Ruby 3.2.0](https://www.ruby-lang.org/en/news/2022/12/25/ruby-3-2-0-released/)
+* [SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html)
+* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html)
+* [Go](https://go.dev/doc/install)
+* [Docker](https://docs.docker.com/get-docker)
+
+
+## Building Lambda Ruby Layer With OpenTelemetry Ruby Dependencies
+
+Build
+```bash
+sam build -u -t template.yml
+```
+
+Make sure the layer structure like below with your zip
+```
+ruby
+    └── gems
+        └── 3.2.0
+            ├── build_info
+            ├── doc
+            ├── extensions
+            ├── gems
+            ├── plugins
+            └── specifications
+```
+
+Zip the file for uploading to ruby lambda layer
+
+```bash
+cd .aws-sam/build/OTelLayer/
+zip -qr ../../../<your_layer_name>.zip ruby/
```

Comment:
Yes, because sam takes the template.yaml file to create folder. However, people can change the name if they change the template.yaml file in [L10](https://github.com/xuan-cao-swi/opentelemetry-lambda/blob/ruby-layer-xuan/ruby/src/template.yml#L10) (e.g. change from OTelLayer to <other_name>), then the folder name will change to <other_name>